src/sail_lean_backend/pretty_print_lean.ml

```diff
@@ -566,22 +566,27 @@ let string_of_type_def_con (TD_aux (td, _)) =
 let doc_typdef ctx (TD_aux (td, tannot) as full_typdef) =
   match td with
   | TD_enum (Id_aux (Id id, _), fields, _) ->
-      let derivers = if List.length fields > 0 then [string "Inhabited"] else [] in
       let fields = List.map doc_id_ctor fields in
       let fields = List.map (fun i -> space ^^ pipe ^^ space ^^ i) fields in
+      let derivers =
+        if List.length fields == 0 then [string "DecidableEq"] else [string "Inhabited"; string "DecidableEq"]
```

Comment:
Should the `then` case be tested?

Reply:
I put that in just to be safe but I'm not sure if it's possible to define empty enums? @Alasdair ?

Reply:
I see. Happy to leave it like this or just making it an assert to catch this case.

Reply:
In general we don't have empty types in Sail (we might even make `--strict-bitvector` the default at some point).  Empty enums are explicitly disallowed; syntactically for normal ones, and by a check for scattered ones.

Reply:
That said, I'm happy to merge this as-is; it's not wrong...

Reply:
Cool. Then let's do that.